### PR TITLE
ODP changes

### PR DESCRIPTION
**Pull request by sumitnigamcs**
@tivv @rmstar @itsankit-google Please review.

### CODE REVIEW

File: src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java [hunk 1]
```diff
@@ -60,4 +60,13 @@
...
   public static WebElement cmekKey;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy=\"datasetProject\" and @class='MuiInputBase-input']")
+  public static WebElement dataSetProjectID;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy=\"dataset\" and @type='text']")
+  public static WebElement dataSet;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy=\"serviceFilePath\" and @type='text']")
+  public static WebElement filePathJson;
```
Comment (itsankit-google) [inline src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java]: filePathJson -> serviceFilePath

**Review by rmstar — COMMENTED** [review]
how is this related to ODP?

File: src/main/java/io/cdap/e2e/pages/actions/CdfSysAdminActions.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 package io.cdap.e2e.pages.actions;
 
+import com.sun.org.apache.xpath.internal.operations.Bool;
```
Comment (rmstar) [inline src/main/java/io/cdap/e2e/pages/actions/CdfSysAdminActions.java]: Remove unnecessary import

**Comment by tivv** [comment]
It looks like there is some framework issue:
```
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
POST https://storage.googleapis.com/storage/v1/b?project=auto-detectcdf-athena&projection=full
{
  "code" : 400,
  "errors" : [ {
    "domain" : "global",
    "message" : "Unknown project id: auto-detectcdf-athena",
    "reason" : "invalid"
  } ],
  "message" : "Unknown project id: auto-detectcdf-athena"
}
```

File: src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java [hunk 3]
```diff
@@ -40,6 +41,9 @@
   public static void enterBigQueryProperties(String tableProp) throws InterruptedException, IOException {
     CdfStudioLocators.bigQueryProperties.click();
     CdfBigQueryPropertiesLocators.bigQueryReferenceName.sendKeys(AUTOMATION_TEST);
+    for (int i=0;i<30;i++) {
```
Comment (itsankit-google) [inline src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java]: Rather than using for loop to clear the value of the element. Can we use something like `CdfBigQueryPropertiesLocators.projectID.clear()`? Do we face any issues with `clear()`? Is there any better way to do it @rmstar, @tivv ?
To be on the safe side, we can do something like this,
```
CdfBigQueryPropertiesLocators.projectID.clear();
while(!CdfBigQueryPropertiesLocators.projectID.getAttribute("value").equals("")) 
{
     CdfBigQueryPropertiesLocators.projectID.sendKeys(Keys.BACK_SPACE);
}
```

Comment (itsankit-google) [reply src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java:44]: I found this workaround and looks much better, we can test this!
`CdfBigQueryPropertiesLocators.projectID.sendKeys(Keys.chord(Keys.CONTROL,"a", Keys.DELETE));`

Comment (rmstar) [reply src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java]: Yes, pressing backspace 30 times is hacky. @itsankit-google your solution looks like a better option if clear() doesn't work.

Comment (rmstar) [reply src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java]: btw, if clear() doesn't work it would be good to document the reason in a comment for future reference.

Comment (sumitnigamcs) [reply src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java]: Ankit suggested to update the method here with the solution: https://github.com/cdapio/cdap-e2e-tests/blob/effa23140514bfbba8cb66e13098d1f0ecf4450d/src/main/java/io/cdap/e2e/utils/SeleniumHelper.java#L126

File: src/main/java/io/cdap/e2e/pages/actions/CdfSysAdminActions.java (hunk 2 above)
Comment (ksahilCF) [reply src/main/java/io/cdap/e2e/pages/actions/CdfSysAdminActions.java]: Removed

File: src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java (hunk 3 above)
Comment (ksahilCF) [reply src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java]: SeleniumHelper has the link documented. Modified it to while loop as suggested 
 public static void replaceElementValue(WebElement element, String value) {
    while (!element.getAttribute("value").equals("")) {
      element.sendKeys(Keys.BACK_SPACE);
    }
    element.sendKeys(value);
  }

File: src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java [hunk 4]
```diff
@@ -60,4 +60,13 @@
 
   @FindBy(how = How.XPATH, using = "//*[@data-cy=\"cmekKey\" and @class='MuiInputBase-input']")
   public static WebElement cmekKey;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy=\"datasetProject\" and @class='MuiInputBase-input']")
+  public static WebElement dataSetProjectID;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy=\"dataset\" and @type='text']")
+  public static WebElement dataSet;
```
Comment (itsankit-google) [inline src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java]: dataSet->dataset

File: src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java [hunk 5]
```diff
@@ -60,4 +60,13 @@
 
   @FindBy(how = How.XPATH, using = "//*[@data-cy=\"cmekKey\" and @class='MuiInputBase-input']")
   public static WebElement cmekKey;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy=\"datasetProject\" and @class='MuiInputBase-input']")
+  public static WebElement dataSetProjectID;
```
Comment (itsankit-google) [inline src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java]: dataSetProjectID->datasetProjectID

File: src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java (hunk 4 above)
Comment (sumitnigamcs) [reply src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java]: done

File: src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java (hunk 5 above)
Comment (sumitnigamcs) [reply src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java]: done

File: src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java [hunk 6]
```diff
@@ -40,7 +39,9 @@
   public static void enterBigQueryProperties(String tableProp) throws InterruptedException, IOException {
     CdfStudioLocators.bigQueryProperties.click();
     CdfBigQueryPropertiesLocators.bigQueryReferenceName.sendKeys(AUTOMATION_TEST);
-    CdfBigQueryPropertiesLocators.projectID.sendKeys(SeleniumHelper.readParameters(PROJECT_ID));
+    SeleniumHelper.replaceElementValue(CdfBigQueryPropertiesLocators.projectID,
+                                       SeleniumHelper.readParameters(PROJECT_ID));
+    CdfBigQueryPropertiesLocators.dataSetProjectID.sendKeys(SeleniumHelper.readParameters(PROJECT_ID));
```
Comment (itsankit-google) [inline src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java]: dataSetProjectID->datasetProjectID

Please run the build locally before submitting, this issue would have been caught there!

File: src/main/java/io/cdap/e2e/utils/GcpClient.java [hunk 7]
```diff
@@ -53,7 +53,7 @@ public static void dropBqQuery(String table) throws IOException, InterruptedExce
     executeQuery(dropQuery);
   }
 
-  private static int executeQuery(String query) throws InterruptedException, IOException {
+  public static int executeQuery(String query) throws InterruptedException, IOException {
```
Comment (itsankit-google) [inline src/main/java/io/cdap/e2e/utils/GcpClient.java:56]: Why this change has been introduced?

Comment (ksahilCF) [reply src/main/java/io/cdap/e2e/utils/GcpClient.java:56]: To execute query other than listed in CDAP framework, we need to call this method. I need to execute where query in ODP but for that we need to make it public.

File: src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java [hunk 8]
```diff
@@ -19,9 +19,8 @@
 import io.cdap.e2e.pages.locators.CdfBigQueryPropertiesLocators;
 import io.cdap.e2e.pages.locators.CdfStudioLocators;
 import io.cdap.e2e.utils.SeleniumHelper;
-
+import org.openqa.selenium.Keys;
```
Comment (itsankit-google) [inline src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java]: Is this required now? If not, please remove the unused import.

Comment (sumitnigamcs) [reply src/main/java/io/cdap/e2e/pages/actions/CdfBigQueryPropertiesActions.java]: done

File: src/main/java/io/cdap/e2e/utils/CdfHelper.java [hunk 9]
```diff
@@ -100,12 +100,12 @@ default void succeedPipelineVerification() {
 
   default int recordIn() {
     String incount = CdfPipelineRunLocators.recordsInCount.getText();
-    return Integer.parseInt(incount);
+    return Integer.valueOf(incount.replaceAll(",", "").toString());
```
Comment (rmstar) [inline src/main/java/io/cdap/e2e/utils/CdfHelper.java]: why do we need to make this change?

Comment (sumitnigamcs) [reply src/main/java/io/cdap/e2e/utils/CdfHelper.java]: we are removing comma from the count value

**Review by itsankit-google — APPROVED** [review]

File: src/main/java/io/cdap/e2e/utils/CdfHelper.java (hunk 9 above)
Comment (itsankit-google) [reply src/main/java/io/cdap/e2e/utils/CdfHelper.java]: don't need to change valueOf->parseInt.

Comment (sumitnigamcs) [reply src/main/java/io/cdap/e2e/utils/CdfHelper.java]: done

**Review by rmstar — APPROVED** [review]
